Console.java

```diff
@@ -0,0 +1,191 @@
+            "commit           Commit the transaction changes and close\n" +
+            "rollback         Rollback the transaction to the beginning state\n" +
+            "close            Close the transaction without committing changes\n" +
+            "<query>          Run graql queries\n" +
+            "source <file>    Run graql queries in file\n";
+
+    private final CommandLineOptions options;
+    private final LineReader reader;
+    private final Printer printer;
+
+    public Console(CommandLineOptions options, LineReader reader, Printer printer) {
+        this.options = options;
+        this.reader = reader;
+        this.printer = printer;
+    }
+
+    public void run() {
+        printer.info(COPYRIGHT);
+        try (Grakn.Client client = new GraknClient(options.address())) {
+            runRepl(client);
+        } catch (GraknClientException e) {
+            printer.error(e.getMessage());
+        }
+    }
+
+    private void runRepl(Grakn.Client client) {
+        while (true) {
+            ReplCommand command = ReplCommand.getCommand(reader, printer, "> ");
+            if (command instanceof ReplCommand.Exit) {
+                break;
+            } else if (command instanceof ReplCommand.Help) {
+                printer.info(HELP_MENU);
+            } else if (command instanceof ReplCommand.Clear) {
+                reader.getTerminal().puts(InfoCmp.Capability.clear_screen);
+            } else if (command instanceof ReplCommand.DatabaseList) {
+                try {
+                    client.databases().all().forEach(database -> printer.info(database));
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.DatabaseCreate) {
+                try {
+                    client.databases().create(command.asDatabaseCreate().database());
+                    printer.info("Database '" + command.asDatabaseCreate().database() + "' created");
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.DatabaseDelete) {
+                try {
+                    client.databases().delete(command.asDatabaseDelete().database());
+                    printer.info("Database '" + command.asDatabaseDelete().database() + "' deleted");
+                } catch (GraknClientException e) {
+                    printer.error(e.getMessage());
+                }
+            } else if (command instanceof ReplCommand.Transaction) {
+                String database = command.asTransaction().database();
+                Grakn.Session.Type sessionType = command.asTransaction().sessionType();
+                Grakn.Transaction.Type transactionType = command.asTransaction().transactionType();
+                boolean shouldExit = runTransactionRepl(client, database, sessionType, transactionType);
```

Comment:
Here we need to return a boolean to tell the parent REPL that the transaction repl has requested console exit, so that it will break out the console loop.

Note that if we do `System.exit(0)` from transaction repl, it will cause an `11:40:02.827 [ForkJoinPool-1-worker-12] ERROR grakn.core.server.rpc.TransactionStream - CANCELLED: cancelled before receiving half close` error on server.